### PR TITLE
topology: remove side effects from macro definitions

### PR DESCRIPTION
**Pull request by Potochi**
This commit fixes the issue mentioned in #4583.

A temporary defition is introduced that takes it's value from
"CHANNELS_MIN" if it is already defined, otherwise it is set to
a default value. This temporary is used instead of "CHANNELS_MIN"
for the rest of the file. This avoids having the macro defined in
files where it should't be.

In order to be completely sure it avoids side effects this temporary
is undefined after it is no longer needed.

Signed-off-by: Bud Liviu-Alexandru <budliviu@gmail.com>

### CODE REVIEW

File: tools/topology/topology1/sof/pipe-volume-playback.m4 [hunk 1]
```diff
@@ -75,14 +75,15 @@ P_GRAPH(pipe-volume-playback, PIPELINE_ID,
 indir(`define', concat(`PIPELINE_SOURCE_', PIPELINE_ID), N_BUFFER(1))
 indir(`define', concat(`PIPELINE_PCM_', PIPELINE_ID), Passthrough Playback PCM_ID)
 
-ifdef(`CHANNELS_MIN',`define(TCHANNELS_MIN, `CHANNELS_MIN')',
-`define(TCHANNELS_MIN, `2')')
+ifdef(`CHANNELS_MIN',`define(`_CHANNELS_MIN', `CHANNELS_MIN')',
```
Comment (lgirdwood) [inline tools/topology/topology1/sof/pipe-volume-playback.m4:78]: Looks like this has already been fixed in this file, but it's missing the undef. Can we rename ```_CHANNELS_MIN``` as  ```LOCAL_CHANNELS_MIN``` as this is easier to follow.
This is not a problem for topology 2.0

**Review by lgirdwood — CHANGES_REQUESTED** [review]
Just need a minor readability update.

File: tools/topology/topology1/sof/pipe-volume-playback.m4 (hunk 1 above)
Comment (dbaluta) [reply tools/topology/topology1/sof/pipe-volume-playback.m4:78]: Yes, here it was already fixed. But indeed renaming TCHANNELS_MIN to _CHANNELS_MIN makes sense.

**Review by iuliana-prodan — APPROVED** [review]

**Review by dbaluta — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
CI reporting known pending zephyr work.

**Pull request merged**